### PR TITLE
add bsp for stm32l4r9

### PR DESCRIPTION
**Pull request by jhbdream**
## 拉取/合并请求描述：(PR description)

[
add touch driver for stm32l4r9-st-eval board
add sram driver for stm32l4r9-st-eval board
fix [HAL_Drivers/drv_gpio.c] for L4 serials about VDDIO2 enable
fix [HAL_Drivers/SConscript] for can not auto remove [soft_iic.c] file when not define BSP_USING_IICx

在STM32L4R9-ST-EVAL 板卡测试通过
]

以下的内容不应该在提交PR时的message修改，修改下述message，PR会被直接关闭。请在提交PR后，浏览器查看PR并对以下检查项逐项check，没问题后逐条在页面上打钩。
The following content must not be changed in submitted PR message. Otherwise, the PR will be closed immediately. After submitted PR, please use web browser to visit PR, and check items one by one, and ticked them if no problem.

### 当前拉取/合并请求的状态 Intent for your PR

必须选择一项 Choose one (Mandatory):

- [ ] 本拉取/合并请求是一个草稿版本 This PR is for a code-review and is intended to get feedback
- [x] 本拉取/合并请求是一个成熟版本 This PR is mature, and ready to be integrated into the repo

### 代码质量 Code Quality：

我在这个拉取/合并请求中已经考虑了 As part of this pull request, I've considered the following:

- [x] 已经仔细查看过代码改动的对比 Already check the difference between PR and old code
- [x] 代码风格正确，包括缩进空格，命名及其他风格 Style guide is adhered to, including spacing, naming and other style
- [x] 没有垃圾代码，代码尽量精简，不包含`#if 0`代码，不包含已经被注释了的代码 All redundant code is removed and cleaned up
- [ ] 所有变更均有原因及合理的，并且不会影响到其他软件组件代码或BSP All modifications are justified and not affect other components or BSP
- [x] 对难懂代码均提供对应的注释 I've commented appropriately where code is tricky
- [x] 本拉取/合并请求代码是高质量的 Code in this PR is of high quality

### CODE REVIEW

**Comment by qgyhd1234** [comment]
更新下 .c 文件的最顶端的注释，日期，作者和说明

**Comment by jhbdream** [comment]
.c 文件 基本没有我自己的修改 @qgyhd1234

**Comment by jhbdream** [comment]
@BernardXiong  这个是重新整理之后的提交

**Comment by jhbdream** [comment]
@BernardXiong 检查并且修改了格式

File: bsp/stm32/stm32l4r9-st-eval/board/board.c [hunk 1]
```diff
@@ -10,50 +10,63 @@
...
+/**
+  * @brief System Clock Configuration
+  * @retval None
+  */
 void SystemClock_Config(void)
 {
-  RCC_OscInitTypeDef RCC_OscInitStruct = {0};
-  RCC_ClkInitTypeDef RCC_ClkInitStruct = {0};
-  RCC_PeriphCLKInitTypeDef PeriphClkInit = {0};
+	RCC_OscInitTypeDef RCC_OscInitStruct = {0};
```
Comment (BernardXiong) [inline bsp/stm32/stm32l4r9-st-eval/board/board.c]: All of indentation is 'TAB'.

**Pull request merged**